File: java/tcso.java
/* dont forget to use throws java.io.IOException in functions
* eg: public static void main(String[] args) throws java.io.IOException { TCSO }
*/ 
try {
 // Try something wrong hee
} catch (Exception e) {
 String sourl;
 sourl = "http://www.stackoverflow.com/search?q=[java]+" + e.getClass();
 sourl = sourl.replace(" ", "+");
 if (java.awt.Desktop.isDesktopSupported()) {
  try {
   java.awt.Desktop.getDesktop().browse(new java.net.URI(sourl));
  } catch (java.net.URISyntaxException ex) {
   java.lang.System.exit(1);
  }
 } else {
  Runtime runtime = Runtime.getRuntime();
  runtime.exec("xdg-open " + sourl);

 }
}

File: scala/tcso.scala
try {
	// doing something stupid
	List().head
} catch {
	case e: Exception => {
		val url = if(e.getMessage != null) {
			"https://stackoverflow.com/search?q=[scala]+" + e.getMessage.replace(" ", "+")
		} else {
			"https://stackoverflow.com/search?q=[scala]+" + e.getClass.getName
		}
		try{
			java.awt.Desktop.getDesktop.browse(new java.net.URI(url))			
		} catch {
			case _ : Throwable => Runtime.getRuntime.exec("xdg-open " + url)
			// if this fails, amen
		}
	}
}
